src/super_gradients/modules/qarepvgg_block.py

```diff
@@ -0,0 +1,305 @@
+        se_type: Type[nn.Module] = nn.Identity,
+        se_kwargs: Union[Mapping[str, Any], None] = None,
+        build_residual_branches: bool = True,
+        use_residual_connection: bool = True,
+        use_alpha: bool = False,
+        use_1x1_bias: bool = True,
+    ):
+        """
+
+        :param in_channels: Number of input channels
+        :param out_channels: Number of output channels
+        :param activation_type: Type of the nonlinearity
+        :param se_type: Type of the se block (Use nn.Identity to disable SE)
+        :param stride: Output stride
+        :param dilation: Dilation factor for 3x3 conv
+        :param groups: Number of groups used in convolutions
+        :param activation_kwargs: Additional arguments for instantiating activation module.
+        :param se_kwargs: Additional arguments for instantiating SE module.
+        :param build_residual_branches: Whether to initialize block with already fused parameters (for deployment)
+        :param use_residual_connection: Whether to add input x to the output (Enabled in RepVGG, disabled in PP-Yolo)
+        :param use_alpha: If True, enables additional learnable weighting parameter for 1x1 branch (PP-Yolo-E Plus)
+        :param use_1x1_bias: If True, enables bias in the 1x1 convolution, authors don't mention it specifically
+        """
+        super().__init__()
+
+        if activation_kwargs is None:
+            activation_kwargs = {}
+        if se_kwargs is None:
+            se_kwargs = {}
+
+        self.groups = groups
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.stride = stride
+        self.dilation = dilation
+        self.activation_type = activation_type
+        self.activation_kwargs = activation_kwargs
+        self.se_type = se_type
+        self.se_kwargs = se_kwargs
+        self.use_residual_connection = use_residual_connection
+        self.use_alpha = use_alpha
+        self.use_1x1_bias = use_1x1_bias
+
+        self.nonlinearity = activation_type(**activation_kwargs)
+        self.se = se_type(**se_kwargs)
+
+        self.branch_3x3 = nn.Sequential()
+        self.branch_3x3.add_module(
+            "conv",
+            nn.Conv2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=3,
+                stride=stride,
+                padding=dilation,
+                groups=groups,
+                bias=False,
+                dilation=dilation,
+            ),
+        )
+        self.branch_3x3.add_module("bn", nn.BatchNorm2d(num_features=out_channels))
+
+        self.branch_1x1 = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=1,
+            stride=stride,
+            padding=0,
+            groups=groups,
+            bias=use_1x1_bias,
+        )
+
+        if use_residual_connection and out_channels == in_channels and stride == 1:
```

Comment:
please assert that out_channels == in_channels if use_residual_connection. 
this is a silent fail. if you use `use_residual_connection=True` and `out_channels != in_channels` or `stride != 1`
this will work unexpectedly